packages/flutter_tools/test/general.shard/ios/simulators_test.dart

```diff
@@ -39,10 +39,12 @@ class MockPlistUtils extends Mock implements PlistParser {}
 
 void main() {
   FakePlatform osx;
+  FileSystemUtils fsUtils;
 
   setUp(() {
     osx = FakePlatform.fromPlatform(const LocalPlatform());
     osx.operatingSystem = 'macos';
+    fsUtils = FileSystemUtils(fileSystem: MemoryFileSystem(), platform: osx);
```

Comment:
Maybe also inject this MemoryFileSystem along with this fsUtils, just to be safe.

Reply:
That sounds good, I also have this https://github.com/flutter/flutter/pull/51671 which should really clean up this file and catch some more global leakage.

Reply:
That sounds good. FYI I also have [this](https://github.com/flutter/flutter/pull/51916) which should clean up this file a lot, and catches a lot of global leakage.

Reply:
But I'll let you land this first, @dnfield.

Reply:
Done - also ProcessManager, since we assert that you have that in there if you have the FS overridden.